internal/spec/spec.go

```diff
@@ -181,30 +187,56 @@ func ToValidLabel(input string) string {
 }
 
 func CreateRunnerVolume(pod *corev1.Pod) error {
-	runnerVolumeName := "runner"
-	runnerVolumeMountPath := "/runner"
-	runnerVolumeEmptyDir := &corev1.EmptyDirVolumeSource{}
-
 	if len(pod.Spec.Containers) < 1 {
 		return fmt.Errorf("pod %s has no runner container spec", pod.Name)
 	}
 
-	runnerContainer := &pod.Spec.Containers[0]
+	// Skip volume creation if a volume with the default name already exists in podTemplate
+	for _, vol := range config.Config.PodTemplate.Spec.Volumes {
+		if vol.Name == runnerVolumeName {
+			return nil
+		}
+	}
 
 	volume := corev1.Volume{
 		Name: runnerVolumeName,
 		VolumeSource: corev1.VolumeSource{
 			EmptyDir: runnerVolumeEmptyDir,
 		},
 	}
+
 	pod.Spec.Volumes = append(pod.Spec.Volumes, volume)
 
+	return nil
+}
+
+func CreateRunnerVolumeMount(pod *corev1.Pod, runnerContainerName string) error {
+	if len(pod.Spec.Containers) < 1 {
+		return fmt.Errorf("pod %s has no runner container spec", pod.Name)
+	}
+
+	// Skip volumemount creation if a volumemount with the same path already exists
+	// in podTemplate for the default container
+	for _, container := range config.Config.PodTemplate.Spec.Containers {
+		if container.Name == runnerContainerName {
+			for _, volMounts := range container.VolumeMounts {
+				// Volumemount paths e.g. /runner and /runner/ are threated equal
+				// The last one in the pod spec will take precedence, which can lead to unexpected behavior
+				if volMounts.MountPath == runnerVolumeMountPath || volMounts.MountPath == runnerVolumeMountPath+"/" {
```

Comment:
```suggestion
				if volMounts.MountPath == filepath.Clean(runnerVolumeMountPath) {
```

by using the `path/filepath` pkg from the standard lib (https://pkg.go.dev/path/filepath#Clean)

Reply:
example snippet: https://go.dev/play/p/nR8FvWytZRD